### PR TITLE
update systemd article to reference dropin file

### PR DESCRIPTION
**Pull request by jessfraz**
let's try to do this without a bike shed please

this PR addresses https://github.com/docker/docker/pull/14513 and all the other PRs about systemd environment files.

ping @tianon @moxiegirl @thaJeztah

### CODE REVIEW

File: docs/articles/systemd.md [hunk 1]
```diff
@@ -33,17 +33,30 @@ If you want Docker to start at boot, you should also:
 There are a number of ways to configure the daemon flags and environment variables
 for your Docker daemon.
 
-If the `docker.service` file is set to use an `EnvironmentFile`
-(often pointing to `/etc/sysconfig/docker`) then you can modify the
-referenced file.
+The recommended way is to use a systemd drop-in file. These are local files in
+the `/etc/systemd/system/docker.service.d` directory.
```
Comment (tianon) [inline docs/articles/systemd.md]: Do you think it's also worthwhile to mention that this is often also just an individual `/etc/systemd/system/docker.service` file? (and that the file works just as well as a drop-in?)

Comment (jfrazelle) [inline docs/articles/systemd.md]: totes

**Comment by thaJeztah** [comment]
:+1: thanks Jess, I'll leave the technical side to you and @tianon for now, but "LGTM" in general

**Comment by moxiegirl** [comment]
LGTM no bikes...not even a tricycle.  There was a unicycle but not worth the change.

File: docs/articles/systemd.md [hunk 2]
```diff
@@ -33,17 +33,32 @@ If you want Docker to start at boot, you should also:
...
+`/etc/systemd/system/docker.service`, which also works for overriding the
+defaults from `/lib/systemd/system/docker.service`.
 
-Check if the `docker.service` uses an `EnvironmentFile`:
+However, if you had previously used a package which had an `EnvironmentFile`
+(often pointing to `/etc/sysconfig/docker`) then for backwards compatibility,
+you drop a file in the `/etc/systemd/system/docker.service.d`
+directory including the following:
+
+    EnvironmentFile=-/etc/sysconfig/docker
```
Comment (eolamey) [inline docs/articles/systemd.md]: Shouldn't the file start with a `[Service]` instruction?

**Comment by jfrazelle** [comment]
updated w a couple changes :)

**Comment by thaJeztah** [comment]
LGTM, feel free to merge if you're done, thanks!

**Comment by tianon** [comment]
LGTM :+1:

**Comment by thaJeztah** [comment]
Thanks!

ping @calavera for cherry pick (you must be getting annoyed by now for my pings ... sorry!)

**Comment by calavera** [comment]
Cherry picked into #15091 

@thaJeztah no problem. Actually, if I'm not pinged I might skip something, so I really appreciate it.

**Pull request merged**